CHANGELOG.md

```diff
@@ -4,6 +4,82 @@ All notable changes to Router will be documented in this file.
+directive @requiresScopes(scopes: [[federation__Scope!]!]!) on OBJECT | FIELD_DEFINITION | INTERFACE | SCALAR | ENUM
+
+directive @authenticated on OBJECT | FIELD_DEFINITION | INTERFACE | SCALAR | ENUM
+```
+
+The implementation hooks into the request lifecycle at multiple steps:
+- In query analysis, we extract the list of scopes necessary to authorize the query.
+- In a supergraph plugin, we calculate the authorization status and put it in the request context:
+    - for `@requiresScopes`, this is the intersection of the query's required scopes and the scopes provided in the request token
+    - for `@authenticated`, it is `is_authenticated` or not
+- In the query planning phase, we filter the query to remove unauthorized fields before proceeding with query planning.
+- At the subgraph level, if query deduplication is active, the authorization status is used to group queries together.
+- At the execution service level, the response is first formatted according to the filtered query, which removed any unauthorized information, then to the shape of the original query, which propagates nulls as needed.
+- At the execution service level, errors are added to the response indicating which fields were removed because they were not authorized.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/3397 https://github.com/apollographql/router/pull/3662
+
+## 🐛 Fixes
+
+### Update deno, so we can generate docs again ([Issue #3305](https://github.com/apollographql/router/issues/3305))
+
+Router docs failed to build on crates.io because of a documentation compile error in Deno.
+This updates Deno to the latest version, which allows us to generate crates.io documentation again.
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/3626
+
+### Fix config metrics path and test for subscription callbacks ([Issue #3687](https://github.com/apollographql/router/issues/3687))
+
+Detection of subscription callbacks has been fixed for internal Apollo metrics. This has no user facing impact.
+
+By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/3688
+
+### GraphQL response processing must happen under the execution span ([PR #3732](https://github.com/apollographql/router/pull/3732))
+
+Previously, any event in processing would be reported under the supergraph span, or any plugin span (like rhai) happening in between
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/3732
+
+## 🛠 Maintenance
+
+### Uplink connections now reuse reqwest client ([Issue #3333](https://github.com/apollographql/router/issues/3333))
+
+Previously uplink requests created a new reqwest client each time, this may cause CPU spikes especially on OSX.
+A single client will now be shared between requests of the same type. 
+
+By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/3703
```

Comment:
```suggestion
### Uplink connections now reuse reqwest client ([Issue #3333](https://github.com/apollographql/router/issues/3333))

A single HTTP client will now be shared between requests of the same type when making requests to [Apollo Uplink](https://www.apollographql.com/docs/federation/managed-federation/uplink/) to fetch supergraphs, licenses and configuration from Studio.  Previously, such requests created a new HTTP client on each periodic fetch which occasionally resulted in CPU spikes, especially on macOS.

By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/3703
```

Reply:
could you reapply this suggestion? I changed the title line just before that, can't apply the change now

Reply:
done https://github.com/apollographql/router/pull/3734/files#r1314932660